File: engine/rules/__fixtures__/vuln.go
package main
import (
	"crypto/md5"; "crypto/sha1"; "crypto/des"; "crypto/rc4"; "crypto/tls"
	"database/sql"; "fmt"; "html/template"; "os/exec"
)
func tlsf() { _ = &tls.Config{ InsecureSkipVerify: true } }
func cmd(name string) { _ = exec.Command("sh", "-c", "ls "+name) }
func q(db *sql.DB, id string) { _, _ = db.Query(fmt.Sprintf("SELECT * FROM u WHERE id=%s", id)) }
func qe(db *sql.DB, id string) { _, _ = db.Exec(fmt.Sprintf("DELETE FROM u WHERE id=%s", id)) }
func xss(s string) template.HTML { return template.HTML(s) }
func h1() { _ = md5.New() }
func h1b(b []byte) { _ = md5.Sum(b) }
func h2() { _ = sha1.New() }
func cdes(k []byte) { _, _ = des.NewCipher(k) }
func crc4(k []byte) { _, _ = rc4.NewCipher(k) }
func ssrf(base, p string) { _, _ = http.Get(base + p) }
func pt(dir, name string) { _, _ = os.Open(dir + name) }

File: engine/rules/__fixtures__/vuln.js
const cp = require('child_process');
const crypto = require('crypto');
function a(req){ return eval(req.query.code); }
const f = new Function('a','b','return a+b');
setTimeout("doStuff()", 100);
function b(name){ return cp.exec(`ls ${name}`); }
function b2(name){ return cp.execSync('ls ' + name); }
function q1(db, id){ return db.query(`SELECT * FROM u WHERE id=${id}`); }
function q2(db, id){ return db.query('SELECT * FROM u WHERE id=' + id); }
function n1(col, code){ return col.find({ $where: code }); }
function deser(s){ return require('node-serialize').unserialize(s); }
function x1(el, v){ el.innerHTML = v; }
function x2(v){ document.write(v); }
function x3(el, v){ el.insertAdjacentHTML('beforeend', v); }
function h1(s){ return crypto.createHash('md5').update(s).digest('hex'); }
function h2(s){ return crypto.createHash("sha1").update(s).digest('hex'); }
const opts = { rejectUnauthorized: false };
process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0';
function tok(){ return Math.random().toString(36); }
function ssrf(base, p){ return fetch(base + p); }
function ssrf2(base, p){ return axios.get(base + p); }
import * as fs from 'fs';
function pt(dir, name){ return fs.readFileSync(dir + name); }
